### PR TITLE
Fix Darwin config use spectrum with ppc64le

### LINKED ISSUE #433: Bug in Regression tests
I believe there are two issues causing problems, one jsrun uses different parameters to specify the number of ranks, they are not all accounted by test_case.py the python regression testing framework. 
This however does not account for the tests that have been failing on darwin since the buffer-in-one pr was merged. 

https://github.com/lanl/parthenon/pull/359

### PR DESCRIPTION
**Pull request by JoshuaSBrown**
## PR Summary

This pr fixes the default build configuration on Darwin. Openmpi compatibility with running gpus was causing problem spectrum mpi does not appear to have the same problems. Namely the advection convergence and performance tests were failing. 

## PR Checklist

- [x] Code passes cpplint
- [x] New features are documented.
- [x] Code is formatted
- [x] Changes are summarized in CHANGELOG.md
- [x] (@lanl.gov employees) Update copyright on changed files

### CODE REVIEW

**Comment by JoshuaSBrown** [comment]
This should be good to go I've tested locally in my user account, but I'm still installing all the spack modules in the parthenon project space so should wait till I test them before this is merged.

File: cmake/machinecfg/Darwin.cmake
Comment (AndrewGaspar) [inline cmake/machinecfg/Darwin.cmake]:
```suggestion
  elseif(DARWIN_MPI_PACKAGE STREQUAL "smpi")
```

File: cmake/machinecfg/Darwin.cmake
Comment (AndrewGaspar) [inline cmake/machinecfg/Darwin.cmake]:
```suggestion
  if(DARWIN_MPI_PACKAGE STREQUAL "openmpi")
```

File: cmake/TestSetup.cmake
Comment (AndrewGaspar) [inline cmake/TestSetup.cmake]: This looks spammy.

```suggestion
```

**Review by AndrewGaspar — CHANGES_REQUESTED** [review]

**Comment by JoshuaSBrown** [comment]
@AndrewGaspar

**Comment by JoshuaSBrown** [comment]
https://re-git.lanl.gov/eap-oss/parthenon/-/jobs/97386 so this shows the ci now passing on Darwin, which it hasn't been. :)

**Comment by JoshuaSBrown** [comment]
@AndrewGaspar @jlippuner @Yurlungur @pgrete ping ping. :)

**Review by AndrewGaspar — APPROVED** [review]

**Comment by AndrewGaspar** [comment]
> ping ping. :)

You should get that looked at.

**Review by Yurlungur — APPROVED** [review]
Thanks for fixing this.

**Pull request merged**
